utils/gen-styleguide-prod-config.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * This script processes the styleguide.config.js to generate a version for
+ * our production documentation.
+ *
+ * 1. It strips any dev-only docs from the styleguide
+ *    This is any section with the property "private" set to true.
+ *
+ * 2. Adds the design package versions from the package.json of each package so
+ *    that we know what design spec and package versions were last released.
+ *
+ * Running `yarn build:styleguidist` will run this script and then run the
+ * styleguidist build, which will generate the docs.
+ *
+ * Running `yarn start` will skip this script so that developers can see both
+ * exported and development documentation.
+ */
+const fs = require("fs");
+const path = require("path");
+
+const {getComponentFilesFromSection} =  require("./styleguidist-config-utils.js");
+
+const styleguideConfig = require("../styleguide.config.js");
+
+/**
+ * Writes a styleguide configuration to a file with the given path.
+ */
+function writeStyleguideConfig(filepath, config) {
+    const lines = [
+        `// This file is auto-generated by gen-styleguide-prod-config.js`,
+        `// Do not edit this file.`,
+        `const {createConfig, babel, postcss} = require("webpack-blocks");`,
+        ``,
+        `module.exports = {`,
+    ];
+    Object.entries(config).forEach(entry => {
+        // HACK(somewhatabstract): We can't just output the webpackConfig
+        // because we executed some things when we loaded it. We could write a
+        // whole thing that reads the text and parses things out, but we don't
+        // need to. We'll just recreate the executed property here.
+        switch (entry[0]) {
+            case "webpackConfig":
+                lines.push("    webpackConfig: createConfig([babel(), postcss()]),");
+                break;
+
+            case "serverPort":
+                // Skip. We do a different port for prod so that prod and dev servers can coexist, if necessary.
+                break;
+
+            default:
+                lines.push(`    ${entry[0]}: ${JSON.stringify(entry[1])},`);
+                break;
+        }
+    });
+    // Let's specify the production server port.
```

Comment:
I didn't do that because the server port setting doesn't exist by default (we use the default port) so that case won't fire necessarily. We want to set this regardless of whether it already exists in the config.

Reply:
I'll update the commentary accordingly.